packages/forms-web-app/__tests__/unit/controllers/document.test.js

```diff
@@ -52,6 +60,44 @@ describe('controllers/document', () => {
 			expect(mockFetchDocument.body.pipe).toHaveBeenCalledWith(res);
 		});
 
+		it('should redirect s78 if no appeal in session', async () => {
+			delete req.session.appeal;
+
+			getExistingAppeal.mockResolvedValue({
+				id: req.params.appealOrQuestionnaireId,
+				appealType: constants.APPEAL_ID.PLANNING_SECTION_78
+			});
+
+			req.baseUrl = '/document';
+			req.url = `/${req.params.appealOrQuestionnaireId}/doc-id`;
+
+			await getDocument(req, res);
+
+			expect(req.session.loginRedirect).toBe(req.baseUrl + req.url);
+			expect(res.redirect).toHaveBeenCalledWith(
+				`/full-appeal/submit-appeal/enter-code/${req.params.appealOrQuestionnaireId}`
+			);
+		});
+
+		it('should redirect HAS if no appeal in session', async () => {
```

Comment:
If you think that is useful, you could potentially have it in one test expression as follows:

[{ appealType: constants.APPEAL_ID.PLANNING_SECTION_78, url: `/full-appeal/submit-appeal/enter-code/${req.params.appealOrQuestionnaireId}`},
{ appealType: constants.APPEAL_ID.HOUSEHOLDER, url: `/appeal-householder-decision/enter-code/${req.params.appealOrQuestionnaireId}`
}].forEach(({appealType, url}) => {
it(`should redirect ${appealType} if no appeal in session`, async () => {
 			delete req.session.appeal;

 			getExistingAppeal.mockResolvedValue({
 				id: req.params.appealOrQuestionnaireId,
 				appealType
 			});

 			req.baseUrl = '/document';
 			req.url = `/${req.params.appealOrQuestionnaireId}/doc-id`;

 			await getDocument(req, res);

 			expect(req.session.loginRedirect).toBe(req.baseUrl + req.url);
 			expect(res.redirect).toHaveBeenCalledWith(url);
 		});
});


I guess it could help if we add more appeal types as well